src/ZeroLog/LogEvent.cs

```diff
@@ -295,27 +291,36 @@ public void Log()
             _log.Enqueue(this);
         }
 
-        public void WriteToStringBuffer(StringBuffer stringBuffer)
+        public void WriteToStringBuffer(StringBuffer stringBuffer, IList<IntPtr>? keyValuePointers)
         {
+            if (keyValuePointers == null)
+            {
+                keyValuePointers = new List<IntPtr>();
```

Comment:
That's not pretty 😉 

I'd encapsulate this into some kind of `AppendBuffer` (or better-named) class and make it a mandatory parameter. It would also be better to wrap a `List<IntPtr>` instead of an `IList<IntPtr>` to avoid many interface calls.